sharding/proposer/service.go

```diff
@@ -43,42 +44,56 @@ func (p *Proposer) Start() {
 // Stop the main loop for proposing collations.
 func (p *Proposer) Stop() error {
 	log.Info(fmt.Sprintf("Stopping proposer service in shard %d", p.shardID))
+	p.txpoolSub.Unsubscribe()
 	return nil
 }
 
+// proposeCollations listens to the transaction feed submits a collation when one is found
 func (p *Proposer) proposeCollations() {
+	requests := make(chan *types.Transaction)
+	p.txpoolSub = p.txpool.TransactionsFeed().Subscribe(requests)
 
-	// TODO: Receive TXs from shard TX generator or TXpool (Github Issues 153 and 161)
-	var txs []*types.Transaction
-	for i := 0; i < 10; i++ {
-		data := make([]byte, 1024)
-		rand.Read(data)
-		txs = append(txs, types.NewTransaction(0, common.HexToAddress("0x0"),
-			nil, 0, nil, data))
+	for {
```

Comment:
This function body here needs test coverage but I didn't see any examples for testing side effects in the sharding package. Any ideas?

Reply:
Can you elaborate more? What makes this difficult to test?

Reply:
I haven't looked at every test in the `sharding` package but most of them assert the return value of a function. For this method I'd like to stub `createCollation` and check that it gets called when a transaction gets sent to the transaction feed but I don't see any mocking libraries being used in any of the test files.

Reply:
Yeah, I'm not sure. We are facing the same issue in other places. 

I suppose you could call `p.cancel` after a short timeout, but that feels like a hacky solution.

Since this is temporary code, I'm ok without testing this for loop, but this is a pattern we are already having issues within non-temporary code. I will spend some more time thinking about how to test this today.

Reply:
Oh wait this is the non-temporary code.